spec/commands/feed/fetch_one_spec.rb

```diff
@@ -1,94 +1,82 @@
 
       allow(Feed::FindNewStories).to receive(:call).and_return([])
 
-      expect(StoryRepository).not_to receive(:add)
+      allow(described_class).to receive(:fetch_raw_feed).and_return(raw_feed)
+      expect(FeedRepository).to receive(:set_status).with(:green, feed)
 
-      described_class.call(daring_fireball)
+      expect { described_class.call(feed) }.not_to change(feed.stories, :count)
     end
   end
 
   context "when new posts have been added" do
-    let(:now) { Time.zone.now }
-    let(:new_story) { double }
-    let(:old_story) { double }
-
-    let(:fake_feed) do
-      double(last_modified: now, entries: [new_story, old_story])
-    end
+    it "only adds posts that are new" do
+      feed = create(:feed)
+      opts = { published: Time.zone.local(2013, 1, 1) }
+      raw_feed = create_raw_feed(entries: [create_entry, create_entry(opts:)])
 
-    before do
-      allow(Feed::FindNewStories).to receive(:call).and_return([new_story])
-    end
+      allow(described_class).to receive(:fetch_raw_feed).and_return(raw_feed)
 
-    it "only adds posts that are new" do
-      expect(FeedRepository).to receive(:set_status)
-      expect(FeedRepository).to receive(:update_last_fetched)
-      expect(Feedjira).to receive(:parse).and_return(fake_feed)
-      expect(StoryRepository).to receive(:add).with(
-        new_story,
-        daring_fireball
-      )
-      expect(StoryRepository)
-        .not_to receive(:add).with(old_story, daring_fireball)
-
-      described_class.call(daring_fireball)
+      expect { described_class.call(feed) }.to change(feed.stories, :count).by(1)
     end
 
     it "updates the last fetched time for the feed" do
-      expect(FeedRepository).to receive(:set_status)
-      expect(StoryRepository).to receive(:add)
-      expect(Feedjira).to receive(:parse).and_return(fake_feed)
-      expect(FeedRepository).to receive(:update_last_fetched)
-        .with(daring_fireball, now)
+      feed = create(:feed, last_fetched: Time.zone.local(2013, 1, 1))
+      raw_feed = create_raw_feed(last_modified: Time.zone.now)
 
-      described_class.call(daring_fireball)
+      allow(described_class).to receive(:fetch_raw_feed).and_return(raw_feed)
+
+      expect { described_class.call(feed) }
+        .to change { feed.last_fetched }.to(raw_feed.last_modified)
     end
   end
 
   context "feed status" do
     it "sets the status to green if things are all good" do
-      fake_feed =
-        double(last_modified: Time.zone.local(2012, 12, 31), entries: [])
-      expect(Feedjira).to receive(:parse).and_return(fake_feed)
+      feed = create(:feed)
+      raw_feed = create_raw_feed(last_modified: Time.zone.now)
 
-      expect(FeedRepository).to receive(:set_status)
-        .with(:green, daring_fireball)
+      allow(described_class).to receive(:fetch_raw_feed).and_return(raw_feed)
 
-      described_class.call(daring_fireball)
+      expect { described_class.call(feed) }
+        .to change { feed.status }.to("green")
```

Comment:
I think you can use the shorthand syntax here:

```suggestion
        .to change(feed, :status).to("green")
```